ntoskrnl/include/internal/cm_x.h

```diff
@@ -226,6 +226,24 @@ CmpConvertKcbSharedToExclusive(IN PCM_KEY_CONTROL_BLOCK k)
     CmpAcquireKcbLockExclusive(k);
 }
 
+//
+// Ensures the KCB lock can be safely acquired in order to avoid deadlocks
+//
+FORCEINLINE
+BOOLEAN
+CmpCanSafelyAcquireKcbLockExclusive(
+    _In_ PCM_KEY_CONTROL_BLOCK Kcb)
+{
+    if (!ExTryToAcquirePushLockExclusive(&CmpCacheTable[GET_HASH_INDEX(Kcb->ConvKey)].Lock))
```

Comment:
It seems the `GET_HASH_INDEX(Kcb->ConvKey)` could be calculated once, and stored in a local variable to be used in the 3 places here and below.